nucypher/blockchain/eth/interfaces.py

```diff
@@ -404,25 +394,11 @@ def _attach_blockchain_provider(
 
             else:
                 providers = {
-                    "auto": _get_auto_provider,
-                    "ipc": _get_IPC_provider,
-                    "file": _get_IPC_provider,
-                    "ws": _get_websocket_provider,
-                    "wss": _get_websocket_provider,
-                    "http": _get_HTTP_provider,
-                    "https": _get_HTTP_provider,
+                    "http": _get_http_provider,
```

Comment:
Do you think this can be safely removed?  If so the provider lookup construction can be simplified further.

Reply:
If there are only one or two logical paths, structured data may be overkill?

Reply:
I assume this is what you were angling at - https://github.com/nucypher/nucypher/pull/3491/commits/24b548f19269824484d6cadb49c37d2dece1ced2.

While it reduces lines, I don't know if I'm convinced that it is clearer - although it's possible you had a different idea here.